### PR TITLE
Display the Main articles section for site users on the Home page.

### PR DESCRIPTION
**Pull request by SerhiyToroniy**

### CODE REVIEW

File: sports_hub/sports_hub/Controllers/HomeController.cs [hunk 1]
```diff
@@ -27,6 +26,24 @@ public HomeController(ILogger<HomeController> logger, ApplicationContext context
 
         public IActionResult Index()
         {
+            const int mainArticlesCount = 5;
```
Comment (MariaPonomarenko38) [inline sports_hub/sports_hub/Controllers/HomeController.cs:29]: you can remove it, admin can't add more than 5 articles

File: sports_hub/sports_hub/Models/Entities/Navigation/Article.cs [hunk 2]
```diff
@@ -26,9 +26,15 @@ public class Article
         public string Content { get; set; }
         public bool Published { get; set; }
         public bool Show_comments { get; set; }
+        public byte[] ArticleImage { get; set; }
         public Category Category { get; set; }
         [Required]
         public string CategoryName { get; set; }
 
+        [Required]
+        public bool IsMainArticle { get; set; }
```
Comment (MariaPonomarenko38) [inline sports_hub/sports_hub/Models/Entities/Navigation/Article.cs:35]: change IsMainArticle field on `public StatusType StatusMainArticle { get; set; }`
add enum type before Article class `public enum StatusType
    {
        Published,
        Saved,
        None
    }`

File: sports_hub/sports_hub/Models/Entities/Navigation/Article.cs [hunk 3]
```diff
@@ -26,9 +26,15 @@ public class Article
...
+
+        [Required]
+        public DateTime dateTime { get; set; }
     }
```
Comment (MariaPonomarenko38) [inline sports_hub/sports_hub/Models/Entities/Navigation/Article.cs:39]: add field `public bool ShowOnMainPage { get; set; }`

File: sports_hub/sports_hub/Controllers/HomeController.cs [hunk 4]
```diff
@@ -27,6 +26,24 @@ public HomeController(ILogger<HomeController> logger, ApplicationContext context
...
+            var mainArticles = _context.Articles.Where(i => i.Published && i.IsMainArticle).Take(mainArticlesCount).ToList<Article>();
```
Comment (MariaPonomarenko38) [inline sports_hub/sports_hub/Controllers/HomeController.cs:30]: change on this `var mainArticles = _context.Articles.Where(i => i.Published && i.StatusMainArticle==StatusType.Published).Take(mainArticlesCount).ToList<Article>();`

**Review by MariaPonomarenko38 — CHANGES_REQUESTED** [review]

**Review by MariaPonomarenko38 — APPROVED** [review]

File: sports_hub/sports_hub/Controllers/HomeController.cs [hunk 5]
```diff
@@ -27,6 +26,23 @@ public HomeController(ILogger<HomeController> logger, ApplicationContext context
 
         public IActionResult Index()
         {
+            var mainArticles = _context.Articles.Where(i => i.Published && i.StatusMainArticle == StatusType.Published).ToList<Article>();
```
Comment (MariaPonomarenko38) [inline sports_hub/sports_hub/Controllers/HomeController.cs:29]: check if ShowOnMainPage property is true

**Review by MariaPonomarenko38 — CHANGES_REQUESTED** [review]

File: sports_hub/sports_hub/Controllers/HomeController.cs (hunk 5 above)
Comment (MariaPonomarenko38) [reply sports_hub/sports_hub/Controllers/HomeController.cs:29]: `var mainArticles = _context.Articles.Where(i => i.Published && i.StatusMainArticle == StatusType.Published && i.ShowOnMainPage).ToList<Article>();`

**Review by MariaPonomarenko38 — APPROVED** [review]

File: sports_hub/sports_hub/Controllers/HomeController.cs [hunk 6]
```diff
@@ -27,6 +26,23 @@ public HomeController(ILogger<HomeController> logger, ApplicationContext context
...
+                item.ArticleImage = System.IO.File.ReadAllBytes("wwwroot/images/Account/sing_up_in.png");
+            }
+            ViewBag.MainArticleImage = System.IO.File.ReadAllBytes("wwwroot/images/Account/sing_up_in.png");
+
+            ViewBag.MainArticles = mainArticles;
+            ViewBag.MainArticle = mainArticle;
+            ViewBag.MainArticleHeadline = mainArticle.Headline;
+            ViewBag.MainArticleCaption = mainArticle.Caption;
+            ViewBag.MainArticleId = mainArticle.Id;
+            ViewBag.MainArticleCategoryName = mainArticle.CategoryName;
```
Comment (LIUBOMYR-HOLKO) [inline sports_hub/sports_hub/Controllers/HomeController.cs:44]: if mainArticle is null will be an exception.

File: sports_hub/sports_hub/Controllers/HomeController.cs [hunk 7]
```diff
@@ -27,6 +26,23 @@ public HomeController(ILogger<HomeController> logger, ApplicationContext context
...
         {
+            var mainArticles = _context.Articles.Where(i => i.Published && i.StatusMainArticle == StatusType.Published && i.ShowOnMainPage).ToList<Article>();
+            var mainArticle = mainArticles.FirstOrDefault();
+
+            //hard-code is only for testing byte converter
+            foreach (var item in mainArticles)
+            {
+                item.ArticleImage = System.IO.File.ReadAllBytes("wwwroot/images/Account/sing_up_in.png");
+            }
+            ViewBag.MainArticleImage = System.IO.File.ReadAllBytes("wwwroot/images/Account/sing_up_in.png");
```
Comment (LIUBOMYR-HOLKO) [inline sports_hub/sports_hub/Controllers/HomeController.cs:37]: hardcode here. Every article will have the same image

File: sports_hub/sports_hub/Models/Entities/Navigation/Article.cs [hunk 8]
```diff
@@ -26,9 +27,17 @@ public class Article
...
+        public byte[] ArticleImage { get; set; }
         public Category Category { get; set; }
         [Required]
         public string CategoryName { get; set; }
 
+        [Required]
+        public StatusType StatusMainArticle { get; set; }
+
+        [Required]
+        public DateTime dateTime { get; set; }
```
Comment (XashDM) [inline sports_hub/sports_hub/Models/Entities/Navigation/Article.cs:39]: Uppercase

File: sports_hub/sports_hub/wwwroot/js/HomeScript.js [hunk 9]
```diff
@@ -1,112 +1,78 @@
...
-                        html: items.join("")
-                    }).appendTo(obj.parentElement.children[1]);
-                },
-                error: function (req, status, error) {
-                    console.log(error);
-                }
-            });
-        }
-        
+//let news = document.getElementsByClassName('mini-news');
```
Comment (XashDM) [inline sports_hub/sports_hub/wwwroot/js/HomeScript.js:13]: remove

**Review by XashDM — APPROVED** [review]
Fix comments

**Pull request merged**